modules/fundamental/src/sbom/service/sbom.rs

```diff
@@ -109,16 +114,92 @@ impl SbomService {
 
         Ok(match node {
             Some(node) => Some(SbomSummary {
-                id: sbom.sbom_id,
-                hashes: vec![Id::Sha256(sbom.sha256)],
-                document_id: sbom.document_id,
+                head: SbomHead {
+                    id: sbom.sbom_id,
+                    hashes: vec![Id::Sha256(sbom.sha256)],
+                    document_id: sbom.document_id,
+                    name: node.name,
+                    labels: sbom.labels,
+                },
 
-                name: node.name,
                 published: sbom.published,
                 authors: sbom.authors,
 
                 described_by,
-                labels: sbom.labels,
+            }),
+            None => None,
+        })
+    }
+
+    /// turn an (sbom, sbom_node) row into an [`SbomDetails`], if possible
+    async fn build_details(
+        &self,
+        (sbom, node): (sbom::Model, Option<sbom_node::Model>),
+        tx: impl AsRef<Transactional>,
+    ) -> Result<Option<SbomDetails>, Error> {
+        let connection = self.db.connection(&tx);
+
+        let described_by = self
+            .describes_packages(sbom.sbom_id, Paginated::default(), tx.as_ref())
+            .await?
+            .items;
+
+        let relevant_advisory_info = sbom
+            .find_related(sbom_package::Entity)
+            .join(JoinType::Join, sbom_package::Relation::Node.def())
+            .join(JoinType::LeftJoin, sbom_package::Relation::Purl.def())
+            .join(
+                JoinType::LeftJoin,
+                sbom_package_purl_ref::Relation::Purl.def(),
+            )
+            .join(
+                JoinType::LeftJoin,
+                qualified_purl::Relation::VersionedPurl.def(),
+            )
+            .join(JoinType::LeftJoin, versioned_purl::Relation::BasePurl.def())
+            .join(JoinType::Join, base_purl::Relation::PurlStatus.def())
+            .join(JoinType::Join, purl_status::Relation::Status.def())
+            .join(
+                JoinType::LeftJoin,
+                purl_status::Relation::VersionRange.def(),
+            )
+            .join(JoinType::LeftJoin, purl_status::Relation::ContextCpe.def())
+            .join(JoinType::Join, purl_status::Relation::Advisory.def())
+            .join(JoinType::Join, purl_status::Relation::Vulnerability.def())
+            .select_only()
+            .columns_prefixed("advisory", advisory::Column::iter())
```

Comment:
To my understanding this seems to align (or must be aligned) with the `QueryCatcher`. Wouldn't it make sense to have an associated function on that struct, which performs all the `columns_prefixed` calls? So that logic lives in the same file?